docs/csharp/language-reference/keywords/file.md

```diff
@@ -0,0 +1,52 @@
+---
+description: "file access modifier: Declare types whose scope is the file in which it's declared"
+title: "file keyword - C# Reference"
+ms.date: 09/15/2022
+f1_keywords: 
+  - "file_CSharpKeyword"
+  - "file"
+helpviewer_keywords: 
+  - "file keyword [C#]"
+---
+# file (C# Reference)
+
+Beginning with C# 11, the `file` contextual keyword is a type access modifier.
+
+File-scoped access restricts a top-level type's scope and visibility to the file in which it's declared. The `file` accessor modifier will generally be applied to types written by a source generator. File scoped types provide source generators with a convenient way to avoid name collisions among generated types. The `file` access modifier declares a file-scoped type, as in this example:
+
+```csharp
+file class HiddenWidget
+{
+    // implementation
+}
+```
+
+Any types nested within a file based type is also only visible within the file in which it's declared. Other types in an assembly may use the same name as a file-scoped type. Because the file-scoped type is visible only in the file where it's declared, these types don't create a naming collision.
+
+A `file` scoped type can't be the return type or parameter type of any member that is more visible than `file` scope. A file scoped type can't be a field member of a type that has greater visibility than `file` scope. However, a more visible type may implicitly implement a file-scoped interface type.
+
+## Example
+
+The following example shows a public type that uses a file scoped type to provide a worker method. In addition, the public type implements a file-scoped interface implicitly:
```

Comment:
```suggestion
The following example shows a public type that uses a file-scoped type to provide a worker method. In addition, the public type implements a file-scoped interface implicitly:
```

Given that the file-scoped interface is explicitly named in the code that declares the public type, in what sense is it implemented implicitly?

Reply:
I'll add a link to clarify.  You can't use explicit interface implementation, so this would be illegal:

```csharp
int IWidget.ProvideAnswer() => 42;
```

If `IWidget` was a file-scoped interface and the class wasn't a file-scoped class.